migration-tool/src/main/java/software/amazon/awssdk/migration/internal/recipe/S3StreamingResponseToV2.java

```diff
@@ -0,0 +1,182 @@
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.migration.internal.recipe;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JContainer;
+import org.openrewrite.java.tree.JRightPadded;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.java.tree.TypeUtils;
+import org.openrewrite.marker.Markers;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.migration.internal.utils.IdentifierUtils;
+
+@SdkInternalApi
+public class S3StreamingResponseToV2 extends Recipe {
+    private static final JavaType.FullyQualified S3_OBJECT_TYPE =
+        TypeUtils.asFullyQualified(JavaType.buildType("com.amazonaws.services.s3.model.S3Object"));
+    private static final JavaType.FullyQualified S3_GET_OBJECT_RESPONSE_TYPE =
+        TypeUtils.asFullyQualified(JavaType.buildType("software.amazon.awssdk.services.s3.model.GetObjectResponse"));
+
+    private static final JavaType.FullyQualified RESPONSE_INPUT_STREAM_TYPE =
+        TypeUtils.asFullyQualified(JavaType.buildType("software.amazon.awssdk.core.ResponseInputStream"));
+
+    private static final MethodMatcher GET_OBJECT_CONTENT =
+        new MethodMatcher("com.amazonaws.services.s3.model.S3Object getObjectContent()");
+
+    private static final MethodMatcher OBJECT_INPUT_STREAM_METHOD =
+        new MethodMatcher("com.amazonaws.services.s3.model.S3ObjectInputStream *(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "V1 S3Object to V2";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Transform usage of V1 S3Object to V2.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new Visitor();
+    }
+
+    private static class Visitor extends JavaVisitor<ExecutionContext> {
+
+        // Transform an S3Object myObject = ...
+        // to
+        // ResponseInputStream<GetObjectResponse> myObject = ...
+        @Override
+        public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable,
+                                                                ExecutionContext executionContext) {
+            if (TypeUtils.isAssignableTo(S3_OBJECT_TYPE, multiVariable.getType())) {
+                JavaType.Parameterized newType =
+                    new JavaType.Parameterized(null, RESPONSE_INPUT_STREAM_TYPE,
+                                               Collections.singletonList(S3_GET_OBJECT_RESPONSE_TYPE));
+
+                maybeAddS3ResponseImports();
```

Comment:
nit : Do we need the maybe ? I thought we could just do AddS3ResponseImports and possibly specify a comment that explains under what conditions this happens.

Reply:
I prefer to keep the `maybe` since these visitor methods can be called any number of times and it clearly signifies that we only conditionally add them (i.e. if they aren't already present).